docs/architecture/posixfs.md

```diff
@@ -0,0 +1,173 @@
+
+### File Id Resolution
+
+Infinite Scale uses an Id based approach to work with resources, rather than a file path based mechanism. The reason for that is that Id based lookups can be done way more efficient compared to tree traversals, just to name one reason.
+
+The most important component of the Id is a unique file Id that identifies the resource within a space. Ideally the Inode of a file could be used here. However, some file systems re-use inodes which must be avoided. Infinite Scale thus does not use the file Inode, but generates a UUID instead.
+
+ID based lookups utilize an Id cache which needs to be shared between all storageprovider and dataprovider instances. During startup a scan of the whole file tree is performed to detect and cache new entities.
+
+In the future a powerful underlying file system could support Infinite Scale by providing an API that
+
+1. Provides the Id for a given file path referenced resource
+2. Provides the path for a given Id.
+
+These two operations are very crucial for the performance of the entire system.
+
+### User Management
+
+With the requirement that data can be manipulated either through the filesystem or the Infinite Scale system, the question under which uid the manipulation happens is an important question.
+
+There are a few possible ways for user management:
+1. Changes can either be only accepted by the same  user that Infinite Scale is running under, for example the user `ocis`. All manipulations in the filesystem have to be done by, and only by, this user.
+2. Group based: All users who should be able to manipulate files have to be in a unix group. The Infinite Scale user has also to be in there. The default umask in the directory used has to allow group writing all over the place.
+3. Impersonation: Infinite Scale impersonates the user who owns the folder on the file system to mimic the access as the user.
+
+All possibilities have pros and cons for operations.
+
+One for all, it seems reasonable to use LDAP to manage users which is the base for the Infinite Scale IDP as well as the system login system via PAM.
+
+## Advanced Features
+
+Depending on the capabilities of the underlying file system, the Infinite Scale PosixFS can benefit from more advanced funcitonality described here.
+
+### Versioning
+
+If the underlying file system is able to create versions of single resources (imagine a git based file system) this functionality could directly be used by Infinite Scale.
+
+In the current state of the PosixFS, versioning is not supported.
+
+### Trashbin
+
+If the underlying file system handles deleted files in a trash bin that allows restoring of previously removed files, this functionality could directly be used by Infinite Scale.
+
+If not available it will follow the [the Free Desktop Trash specificaton](https://specifications.freedesktop.org/trash-spec/trashspec-latest.html).
+
+In the current state of the PosixFS, trash bin is not supported.
```

Comment:
```suggestion
In the current state of the PosixFS, trash bin is not supported.

Infinite Scale could, if provided by a later version:

- If the underlying file system handles deleted files in a trash bin that allows restoring of previously removed files, this functionality could directly be used by Infinite Scale.
- If not available it will follow the [Free Desktop Trash specificaton](https://specifications.freedesktop.org/trash-spec/trashspec-latest.html).
```

Reply:
same here